### ISSUE TITLE
[BUG] Error on downloading free course

### ISSUE THREAD

**Issue by rbasniak** [nishad/udemy-dl#144]
Version : **2.1.1**

- [X] I have read and understood the [contributors guide](https://github.com/nishad/udemy-dl/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/nishad/udemy-dl/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/nishad/udemy-dl/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

- [ ] I have never used the project.
- [X] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Hi, I found some issues with the same error and they were closed, but I didn't find anything to help me...

I'm trying to download this free course: 

https://www.udemy.com/bootstrap-4

All I get is the following output and nothing downloads:

```
[INFO-881] Downloading to: F:\Udemy Courses\bootstrap-4

[INFO-127] Trying to log in ...
[INFO-157] Login success.
[INFO-189] Found course id: 708578
Getting course data: 11.11% [==------------------] [CRITICAL-425] Cannot download lecture "Whats New in Bootstrap 4": "'NoneType' object has no attribute 'group'"
Getting course data: 16.67% [===-----------------] [CRITICAL-425] Cannot download lecture "How to Download  Install Bootstrap 4": "'NoneType' object has no attribute 'group'"
Getting course data: 22.22% [====----------------] [CRITICAL-425] Cannot download lecture "Final Projects Walk Through  Download Course Files": "'NoneType' object has no attribute 'group'"
Getting course data: 33.33% [=======-------------] [CRITICAL-425] Cannot download lecture "Code a Responsive Website (Part 1)": "'NoneType' object has no attribute 'group'"
Getting course data: 38.89% [========------------] [CRITICAL-425] Cannot download lecture "Code a Responsive Website (Part 2)": "'NoneType' object has no attribute 'group'"
Getting course data: 44.44% [=========-----------] [CRITICAL-425] Cannot download lecture "Code a Responsive Website (Part 3)": "'NoneType' object has no attribute 'group'"
Getting course data: 55.56% [===========---------] [CRITICAL-425] Cannot download lecture "Code a Landing Page with a Bootstrap Template": "'NoneType' object has no attribute 'group'"
Getting course data: 66.67% [=============-------] [CRITICAL-425] Cannot download lecture "Code a Startup Landing Page (Part 1)": "'NoneType' object has no attribute 'group'"
Getting course data: 72.22% [==============------] [CRITICAL-425] Cannot download lecture "Code a Startup Landing Page (Part 2)": "'NoneType' object has no attribute 'group'"
Getting course data: 77.78% [================----] [CRITICAL-425] Cannot download lecture "Code a Startup Landing Page (Part 3)": "'NoneType' object has no attribute 'group'"
Getting course data: 88.89% [==================--] [CRITICAL-425] Cannot download lecture "How to Install  Compile SASS": "'NoneType' object has no attribute 'group'"
Getting course data: 94.44% [===================-] [CRITICAL-425] Cannot download lecture "Professional SASS Organization  Workflow": "'NoneType' object has no attribute 'group'"
Getting course data: 100.00% [====================] Done
[CRITICAL-425] Cannot download lecture "Wrap Up  Whats Next": "'NoneType' object has no attribute 'group'"
[INFO-747] Logging out...
[INFO-749] Logging out success
```

**Comment by andesitedesign** [comment]
I'm having the same issue with two courses that I have paid for. Have you found a solution?

**Comment by rbasniak** [comment]
No solution yet :( However, the paid courses I tried to download mostly worked. Some show this same error in one or two videos but everything else on the course gets downloaded.

**Comment by nishad** [comment]
This issue is a duplicate of https://github.com/nishad/udemy-dl/issues/125. Which is addressed in resent development release.

**Comment by mikeatgithub** [comment]
On October 4th, 2016, I downloaded your udemy-dl-win-2.1.1.zip file which is 31,483KB. I extracted it. Next, I ran update.cmd file/command. Then I attempted to download "Angular 2 with TypeScript" by Mosh Hamedani which I have paid for and wanted it for my own use. I got the same error:

Getting course data: 96.77% [===================-] [CRITICAL-425] Cannot download lecture "lecture name": "'NoneType' object has no attribute 'group'". 

It seems like this is a stubborn bug and it is still persisting even in the most recent version of the windows version at least. I really appreciate all the work you are putting in this effort.
